### PR TITLE
add offline first with graphql repository

### PR DESCRIPTION
**Pull request by tshedor**

### CODE REVIEW

File: example/server.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
+const { ApolloServer } = require("apollo-server");
+
+const typeDefs = gql`
+  type Customer {
+    id: Int
+    firstName: String
+    lastName: String
+    pizzas: [Pizza]
```
Comment (Josiassejod1) [inline example/server.js:8]: Pizza Pizza 🍕

File: example/server.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  }
+
+  type Pizza {
+    id: Int
+    frozen: Boolean
```
Comment (Josiassejod1) [inline example/server.js:13]: alpha

File: packages/brick_offline_first/lib/src/offline_first_with_graphql/offline_first_with_graphql_repository.dart [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+  @override
+  @mustCallSuper
+  Future<void> migrate() async {
+    await super.migrate();
+
+    // Migrate cached jobs schema
+    await offlineRequestQueue.link.requestManager.migrate();
+  }
+
+  /// [throwOnReattemptStatusCodes] - when `true`, the repository will throw an
```
Comment (Josiassejod1) [inline packages/brick_offline_first/lib/src/offline_first_with_graphql/offline_first_with_graphql_repository.dart:120]:
```suggestion
  /// [throwOnReattemptStatusCodes] - when `true`, the repository will throw an
```can delete

File: packages/brick_offline_first/lib/src/offline_first_with_graphql/offline_first_with_graphql_repository.dart [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+  }) async {
+    try {
+      return await super.get(
+        query: query,
+        alwaysHydrate: alwaysHydrate,
+        hydrateUnexisting: hydrateUnexisting,
+        requireRemote: requireRemote,
+        seedOnly: seedOnly,
+      );
+    } on LinkException catch (e) {
```
Comment (Josiassejod1) [inline packages/brick_offline_first/lib/src/offline_first_with_graphql/offline_first_with_graphql_repository.dart:92]: Change this to socket exception

Comment (tshedor) [reply packages/brick_offline_first/lib/src/offline_first_with_graphql/offline_first_with_graphql_repository.dart:92]: A SocketException isn't going to throw here, it'll throw in the OfflineCache

File: packages/brick_graphql/lib/src/transformers/model_fields_document_transformer.dart [hunk 5]
```diff
@@ -105,7 +105,7 @@ class ModelFieldsDocumentTransformer<_Model extends GraphqlModel> {
         selectionSet: entry.value.association && !ignoreAssociations
             ? SelectionSetNode(
                 selections: _generateNodes(
-                  modelDictionary.adapterFor[entry.value.runtimeType]!.fieldsToRuntimeDefinition,
+                  modelDictionary.adapterFor[entry.value.type]!.fieldsToRuntimeDefinition,
```
Comment (Josiassejod1) [inline packages/brick_graphql/lib/src/transformers/model_fields_document_transformer.dart:108]: tricky eel over here 🐍

File: packages/brick_graphql/test/model_fields_document_transformer_test.dart [hunk 6]
```diff
@@ -21,6 +21,7 @@ const upsertPersonWithNodes = r'''mutation UpsertPerson($input: UpsertPersonInpu
   upsertPerson(filter: $input) {
     primaryKey
     id
+    assoc
     someField
     complexFieldName
```
Comment (Josiassejod1) [inline packages/brick_graphql/test/model_fields_document_transformer_test.dart:26]: alphabet boyz

File: packages/brick_offline_first/lib/src/offline_first_with_graphql/offline_first_with_graphql_repository.dart [hunk 7]
```diff
@@ -0,0 +1,149 @@
...
+    bool requireRemote = false,
+    bool seedOnly = false,
+  }) async {
+    try {
+      return await super.get(
+        query: query,
+        alwaysHydrate: alwaysHydrate,
+        hydrateUnexisting: hydrateUnexisting,
+        requireRemote: requireRemote,
+        seedOnly: seedOnly,
```
Comment (Josiassejod1) [inline packages/brick_offline_first/lib/src/offline_first_with_graphql/offline_first_with_graphql_repository.dart:85]: alpha

**Review by Josiassejod1 — APPROVED** [review]
⛏ we have lift off with minor knits

**Pull request merged**